adagrams/game.py

```diff
@@ -1,11 +1,169 @@
+'''
+A program to set up and run the Adagrams Game.
+The goal of the game is to create words drawn from a 
+limitted letter pool. Each word is scored based off 
+of which letters were used and based off of length.
+
+Used the random module in order to selet a random array
+of letters from the LETTER_POOL and add the to the
+letter bank.
+
+Used Counter subclass from the collections module 
+to collect the information on the remaining amount 
+of each letter in the letter bank.
+
+'''
+
+
+import random
+from collections import Counter
+
+LETTER_POOL = {
+    'A': [9, 1], 
+    'B': [2, 3], 
+    'C': [2, 3], 
+    'D': [4, 2], 
+    'E': [12, 1], 
+    'F': [2, 4], 
+    'G': [3, 2], 
+    'H': [2, 4], 
+    'I': [9, 1],
+    'J': [1, 8],
+    'K': [1, 5],
+    'L': [4, 1],
+    'M': [2, 3],
+    'N': [6, 1], 
+    'O': [8, 1],
+    'P': [2, 3],
+    'Q': [1, 10], 
+    'R': [6, 1],
+    'S': [4, 1],
+    'T': [6, 1], 
+    'U': [4, 1], 
+    'V': [2, 4], 
+    'W': [2, 4], 
+    'X': [1, 8], 
+    'Y': [2, 4],
+    'Z': [1, 10]
+}
+
+
 def draw_letters():
-    pass
+    """
+    Creates a list of letters based on frequency table, then randomly selects 10 letters
+
+        Parameters:
+                None
+
+        Returns:
+                returned_letters (list): list of 10 randomly chosen letters
+    """
+    letter_list = []
+    returned_letters = []
+    for letter, number in LETTER_POOL.items():
+        letter_list += letter * number[0]
+    for i in range(10):
+        current_letter = letter_list[random.randint(0, len(letter_list) - 1)]
+        returned_letters.append(current_letter)
+        letter_list.remove(current_letter)
+
+    return returned_letters
+    
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    Checks if the provided word utilizes letters only found in letter bank
+
+        Parameters:
+                word (str): the played word
+                letter_bank (list): bank of randomly assigned letters
+
+        Returns:
+                boolean value based on word validity
+    """
+    letter_dict = Counter(letter_bank)
+
+    for letter in word.upper():
+        try:
+            if letter in letter_dict.keys() and letter_dict[letter] > 0:
```

Comment:
Take care to use `key in dict syntax`. If you do`key in dict.keys()` you'd reproduce O(n) time complexity since under the hood Python creates a list of keys when you call .keys() and the in operator for lists is linear.

`if letter in letter_dict and letter_dict[letter] > 0:`